Source/Shaders/ModelExperimental/FeatureStageVS.glsl

```diff
@@ -0,0 +1,35 @@
+vec2 computeSt(float featureId)
+{
+    float stepX = model_textureStep.x;
+    float centerX = model_textureStep.y;
+
+    #ifdef MULTILINE_BATCH_TEXTURE
+    float stepY = model_textureStep.z;
+    float centerY = model_textureStep.w;
+
+    float xId = mod(featureId, model_textureDimensions.x); 
+    float yId = floor(featureId / model_textureDimensions.x);
+    
+    return vec2(centerX + (xId * stepX), centerY + (yId * stepY));
+    #else
+    return vec2(centerX + (featureId * stepX), 0.5);
+    #endif
+}
+
+vec3 featureStage(vec3 position)
+{
+    float featureId;
+
+    #ifdef FEATURE_ID_TEXTURE
+    featureId = floor(texture2D(FEATURE_ID_TEXTURE, FEATURE_ID_TEXCOORD).FEATURE_ID_CHANNEL * 255.0 + 0.5);
+    #else
+    featureId = FEATURE_ID_ATTRIBUTE;
+    #endif
+
+    if (featureId < model_featuresLength)
+    {
+        model_featureSt = computeSt(featureId);
+    }
```

Comment:
Hm, I wonder if it would be better to call `computeSt()` for every fragment (even if result is invalid), that way we avoid divergence in the shader.

@lilleyse do you have an opinion on this?

Reply:
I took this from @lilleyse's implementation in `feature-id-texture-mvp`. The condition was used in this as well as in the `pickId`. See https://github.com/CesiumGS/cesium/blob/48f6b14e7cba610e3e59587359629189ba6081cb/Source/Scene/Cesium3DTileBatchTable.js#L888-L896

Reply:
The reason for the ternary was to support null feature IDs: https://github.com/CesiumGS/glTF/pull/8

I'm not sure the approach here will work for that. Say the feature id is greater than or equal to `model_featuresLength`, it will still try to read the pick texture at coordinates (0.0, 0.0).

Since the code may change I don't have too much to say about the branching, except some branching somewhere is probably inevitable (either as an if statement or a ternary)